src/model/transaction/__tests__/insertFragmentIntoContentState-test.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+ui_infra
+ */
+
+'use strict';
+
+jest.disableAutomock();
+
+var Immutable = require('immutable');
+var insertFragmentIntoContentState = require('insertFragmentIntoContentState');
+var getSampleStateForTesting = require('getSampleStateForTesting');
+var BlockMapBuilder = require('BlockMapBuilder');
+var ContentBlock = require('ContentBlock');
+
+describe('insertFragmentIntoContentState', () => {
+  var sample = getSampleStateForTesting();
+  var content = sample.contentState;
+  var selection = sample.selectionState;
+
+  var block = content.getBlockMap().first();
+  var data = new Immutable.Map({a: 1});
+  var secondData = new Immutable.Map({b: 2});
+
+  function createFragment() {
+    var fragmentArray = [
+      new ContentBlock({
+        key: 'j',
+        type: 'unstyled',
+        text: 'xx',
+        characterList: Immutable.List(),
+        data: data
+      })
+    ];
+    return BlockMapBuilder.createFromArray(fragmentArray);
+  }
+
+  function createMultiblockFragment() {
+    var fragmentArray = [
+      new ContentBlock({
+        key: 'j',
+        type: 'unstyled',
+        text: 'xx',
+        characterList: Immutable.List(),
```

Comment:
Nitpicking, and this should probably not affect anything in the test, but `characterList` should be populated by `text.length` number of `CharacterMetadata` objects. `CharacterMetadata.EMPTY` would do the trick.